### PR TITLE
[ADD] [13.0] mis_builder_budget_tier_validation

### PR DESCRIPTION
**Pull request by misern2**
Moved from : OCA/mis-builder#272

##  Install

This module depends on ``base_tier_validation``. You can find it at  [OCA/server-ux](https://github.com/OCA/server-ux)

This module extends the functionality of Mis Budgets to support a tier validation process.

## Configuration
To configure this module, you need to:  

- Go to *Settings > Technical > Tier Validations > Tier Definition*.
-    Create as many tiers as you want for Mis Budget model.

## Usage
To use this module, you need to:

* Create a Mis Budget triggering at least one "Tier Definition".
* Click on *Request Validation* button.
* Under the tab *Reviews* have a look to pending reviews and their statuses.
* Once all reviews are validated button to *Confirm* will be shown.

Additional features:

* You can filter the Mis Budget requesting your review through the filter *Needs my
  Review*.
* User with rights to confirm the Mis Budget (validate all tiers that would
  be generated) can directly do the operation, this is, there is no need for
  her/him to request a validation.

### CODE REVIEW

**Review by HaraldPanten — APPROVED** [review]
Functional review again 👍

File: mis_builder_budget_tier_validation/tests/test_tier_validation.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+from odoo.tests import common
+
+from .common import setup_test_model, teardown_test_model
+from .tier_validation_tester import TierValidationTester
+
+
+@common.at_install(False)
+@common.post_install(True)
+class TestPurchaseTierValidation(common.SavepointCase):
```
Comment (LoisRForgeFlow) [inline mis_builder_budget_tier_validation/tests/test_tier_validation.py]: here

File: mis_builder_budget_tier_validation/tests/test_tier_validation.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    @classmethod
+    def setUpClass(cls):
+        super(TestPurchaseTierValidation, cls).setUpClass()
```
Comment (LoisRForgeFlow) [inline mis_builder_budget_tier_validation/tests/test_tier_validation.py]: here

**Review by danimv5 — APPROVED** [review]
Functional review! LGTM

**Review by eduardgm — APPROVED** [review]
Functional review 👍🏻  ,LGTM!

**Review by rescola — APPROVED** [review]
Functional LGTM

File: mis_builder_budget_tier_validation/tests/test_tier_validation.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                "definition_domain": "[('test_field', '>', 1.0)]",
+            }
+        )
+
+        cls.test_record = cls.test_model.create({"test_field": 2.5})
+
+    @classmethod
+    def tearDownClass(cls):
+        teardown_test_model(cls.env, [TierValidationTester])
+        super(TestPurchaseTierValidation, cls).tearDownClass()
```
Comment (LoisRForgeFlow) [inline mis_builder_budget_tier_validation/tests/test_tier_validation.py]: here

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

Very small comment, the test class name can be changed

File: mis_builder_budget_tier_validation/tests/test_tier_validation.py (hunk 1 above)
Comment (misern2) [reply mis_builder_budget_tier_validation/tests/test_tier_validation.py]: done :+1:

File: mis_builder_budget_tier_validation/tests/test_tier_validation.py (hunk 3 above)
Comment (misern2) [reply mis_builder_budget_tier_validation/tests/test_tier_validation.py]: done :+1:

File: mis_builder_budget_tier_validation/tests/test_tier_validation.py (hunk 2 above)
Comment (misern2) [reply mis_builder_budget_tier_validation/tests/test_tier_validation.py]: done :+1:

**Pull request merged**